python-sdk/nuscenes/eval/lidarseg/validate_submission.py

```diff
@@ -0,0 +1,129 @@
+import argparse
+import json
+import os
+
+import numpy as np
+from tqdm import tqdm
+
+from nuscenes import NuScenes
+from nuscenes.eval.lidarseg.utils import get_samples_in_eval_set
+from nuscenes.utils.data_classes import LidarPointCloud
+
+
+def validate_submission(nusc: NuScenes, results_folder: str, eval_set: str, verbose: bool = False) -> None:
+    """
+    Checks if a results folder is valid. The following checks are performed:
+    - Check that the submission folder is of the following structure:
+        └── results_folder
+            ├── lidarseg
+            │   └── test <- Contains the .bin files; a .bin file
+            │               contains the labels of the points in a
+            │               point cloud
+            └── test
+                └── submission.json  <- contains certain information about
+                                        the submission
```

Comment:
Better to reference the structure in the other file above to have a single source of truth.